core/geometry/src/polyface/Polyface.ts

```diff
@@ -316,126 +355,145 @@ export class IndexedPolyface extends Polyface {
+      if (priorIndexB !== undefined && normalIsDuplicate(this.data.normal, priorIndexB))
+        return priorIndexB;
+      // check the tail index for possible duplicate
       if (priorIndexA !== undefined || priorIndexB !== undefined) {
         const tailIndex = this.data.normal.length - 1;
-        distance = this.data.normal.distanceIndexToPoint(tailIndex, normal);
-        if (distance !== undefined && Geometry.isSmallMetricDistance(distance))
+        if (normalIsDuplicate(this.data.normal, tailIndex))
           return tailIndex;
       }
     }
-
     return this.addNormalXYZ(normal.x, normal.y, normal.z);
   }
-
-  /** Add a normal vector given by direct coordinates
-   * @returns 0-based index of the added or reused param.
+  /**
+   * Add a normal vector to the normal array.
+   * @param x the x coordinate of normal.
+   * @param y the y coordinate of normal.
+   * @param z the z coordinate of normal.
+   * @returns zero-based index of the added normal vector.
    */
   public addNormalXYZ(x: number, y: number, z: number): number {
-    if (!this.data.normal) this.data.normal = new GrowableXYZArray();
+    if (!this.data.normal)
+      this.data.normal = new GrowableXYZArray();
     this.data.normal.pushXYZ(x, y, z);
     return this.data.normal.length - 1;
   }
-
-  /** Add a color
-   * @returns 0-based index of the added or reused color.
+  /**
+   * Add a color to the color array
+   * @param color the color.
+   * @returns zero-based index of the added color.
    */
   public addColor(color: number): number {
-    if (!this.data.color) this.data.color = [];
+    if (!this.data.color)
+      this.data.color = [];
     this.data.color.push(color);
     return this.data.color.length - 1;
   }
   /** Add a point index with edge visibility flag. */
-  public addPointIndex(index: number, visible: boolean = true): void { this.data.pointIndex.push(index); this.data.edgeVisible.push(visible); }
-  /** Add a normal index */
+  public addPointIndex(index: number, visible: boolean = true): void {
+    this.data.pointIndex.push(index);
+    this.data.edgeVisible.push(visible);
+  }
+  /** Add a normal index. */
   public addNormalIndex(index: number): void {
     if (!this.data.normalIndex)
       this.data.normalIndex = [];
     this.data.normalIndex.push(index);
   }
-  /** Add a param index */
+  /** Add a param index. */
   public addParamIndex(index: number): void {
     if (!this.data.paramIndex)
       this.data.paramIndex = [];
     this.data.paramIndex.push(index);
   }
-  /** Add a color index */
+  /** Add a color index. */
   public addColorIndex(index: number): void {
     if (!this.data.colorIndex)
       this.data.colorIndex = [];
     this.data.colorIndex.push(index);
   }
-
-  /** clean up the open facet.  return the returnValue (so caller can easily return cleanupOpenFacet("message")) */
+  /**
+   * Clean up the open facet.
+   * @deprecated in 4.x. Its single caller does not need to call this API.
```

Comment:
Lint forced me to provide a reason for deprecate. Not sure if this is best description of the reason!